pkg/apis/logging/v1alpha1/forwarding_types.go

```diff
@@ -0,0 +1,196 @@
+	LogSourceTypeInfra LogSourceType = "logs.infra"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// LogForwading is the Schema for the logforwardings API
+// +k8s:openapi-gen=true
+type LogForwarding struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ForwardingSpec    `json:"spec,omitempty"`
+	Status *ForwardingStatus `json:"status,omitempty"`
+}
+
+//ForwardingSpec specifies log forwarding pipelines from a defined sources to dest outputs
+// +k8s:openapi-gen=true
+type ForwardingSpec struct {
+	DisableDefaultForwarding bool           `json:"disableDefaultForwarding,omitempty"`
+	Outputs                  []OutputSpec   `json:"outputs,omitempty"`
+	Pipelines                []PipelineSpec `json:"pipelines,omitempty"`
+}
+
+//PipelineSpec is the sources spec to named targets
+type PipelineSpec struct {
+	Name       string        `json:"name,omitempty"`
+	SourceType LogSourceType `json:"inputType,omitempty"`
+
+	//OutputRefs is a list of  the names of outputs defined by forwarding.outputs
+	OutputRefs []string `json:"outputRefs,omitempty"`
+}
+
+//OutputSpec specifies destination config for log message endpoints
+type OutputSpec struct {
+	Type     OutputType        `json:"type,omitempty"`
+	Name     string            `json:"name,omitempty"`
+	Endpoint string            `json:"endpoint,omitempty"`
+	Secret   *OutputSecretSpec `json:"secret,omitempty"`
+}
+
+//OutputSecretSpec specifies secrets for pipelines
+type OutputSecretSpec struct {
+
+	//Name is the name of the secret to use with this output
+	Name string `json:"name"`
+}
+
+//OutputType defines the type of endpoint that will receive messages
+type OutputType string
+
+const (
+	//OutputTypeElasticsearch configures pipeline to send messages to elasticsearch
+	OutputTypeElasticsearch OutputType = "elasticsearch"
+
+	//OutputTypeForward configures the pipeline to send messages via Fluent's secure forward
+	OutputTypeForward OutputType = "forward"
+)
+
+type LogForwardingStatus struct {
+	LogForwardingState LogForwardingState `json:"state,omitempty"`
+	Reason             string             `json:"reason,omitempty"`
+}
+
+type LogForwardingState string
+
+const (
+	LogForwardingStateAccepted LogForwardingState = "accepted"
+	LogForwardingStateRejected LogForwardingState = "rejected"
+)
+
+//ForwardingStatus is the status of spec'd forwarding
+// +k8s:openapi-gen=true
+type ForwardingStatus struct {
+	LogForwardingStatus LogForwardingStatus `json:"status,omitempty"`
+
+	//LogSources lists the configured log sources
+	LogSources []string `json:"sources,omitempty"`
+
+	//Pipelines is the status of the outputs
+	Pipelines []PipelineStatus `json:"pipelines,omitempty"`
+
+	//Outputs is the status of the outputs
+	Outputs []OutputStatus `json:"outputs,omitempty"`
+}
+
+//PipelineStatus is the status of a give pipeline
+type PipelineStatus struct {
```

Comment:
can we use `conditions` like other k8s components use? (including our operators)

Reply:
use in status:
https://github.com/openshift/cluster-logging-operator/blob/master/pkg/apis/logging/v1/clusterlogging_types.go#L30

definition:
https://github.com/openshift/cluster-logging-operator/blob/master/pkg/apis/logging/v1/clusterlogging_types.go#L241-L247